plone/app/contentrules/conditions/fileextension.py

```diff
@@ -66,10 +60,6 @@ def __call__(self):
         if IFile.providedBy(obj):
             base_unit = getattr(obj, "file", None)
             name = getattr(base_unit, "filename", None)
-        elif IFileContent is None:
-            return False
-        elif not IFileContent.providedBy(obj):
-            return False
         else:
```

Comment:
This `else` was only executed if `IFileContent.providedBy(obj)`. This can no longer be true, so this code branch can be removed.

Reply:
Hmm, no, its now executed if not IFile.providedBy(obj)